app/routes.py

```diff
@@ -1,2 +1,344 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+import requests
+from dotenv import load_dotenv
+load_dotenv()
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST", "GET"])
+def handle_tasks():
+    if request.method == "GET":
+        task = request.args.get("task")
+        task_query = request.args.get("sort")
+
+        if task_query:
+            if task_query == "asc":
+                tasks = Task.query.order_by(Task.title.asc())
+            elif task_query == "desc":
+                tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append(
+                {
+                    "id" : task.task_id,
+                    "title" : task.title,
+                    "description" : task.description,
+                    "is_complete" : task.is_complete
+                }
+            )
+        
+        return jsonify(tasks_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        #replace with try/except
+        #this is my guard clause
+        if (("title" not in request_body.keys()) or 
+        ("description" not in request_body.keys()) or
+        ("completed_at" not in request_body.keys())):
+            return jsonify(
+                {"details": "Invalid data"}
+            ), 400
+
+        if not request_body["completed_at"]:
```

Comment:
Notice how similar the two halves of this if/else are. They differ only in whether something get's passed for `is_complete`. If we eliminate the `is_complete` column, they would be identical and we could omit the condition. This would involve us calculating the value of `is_complete` on the fly in our dictionary result. So for line 66, we might have
```python
"is_complete" : new_task.is_complete is not None
```